### PR TITLE
[FR] Functions

### PR DESCRIPTION
**Pull request by doomspork**
@tchoutri / @julienXX — Can you check that this is satisfactory?

Thank you again for your help 👍

### CODE REVIEW

File: fr/lessons/basics/functions.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+title: Fonctions
+category: basics
+order: 7
+lang: fr
+---
+
+En Elixir, comme dans tous les langages fonctionnels, les fonctions sont des citoyens de premier ordre. Nous verrons les différents types de fonctions en
+Elixir, ce qui les rend différentes et comment les utiliser.
+
+## Table des matières
```
Comment (tchoutri) [inline fr/lessons/basics/functions.md]: Maybe replace the whole block with 

```
## Table des matières
{% include toc.html %}
```

Comment (doomspork) [inline fr/lessons/basics/functions.md]: Ah, good catch 👍

**Comment by doomspork** [comment]
@tchoutri I updated the ToC, any other changes? 😀

**Comment by tchoutri** [comment]
yeah if you're in a playful mood you can `sed` the file to add a space after each and every `:` character, since they are a double-punctuation (like `;`, `!`, `?`), but I can do it when I'm @ home.

**Comment by doomspork** [comment]
@tchoutri I'm not sure I follow, could you point out an instance on this PR?

File: fr/lessons/basics/functions.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+Elixir, ce qui les rend différentes et comment les utiliser.
+
+{% include toc.html %}
+
+## Fonctions anonymes
+
+Tout comme leur nom le sous-entend, les fonctions anonymes n'ont pas de nom. Tel que nous l'avons vu dans la leçon `Enum`, elles sont fréquemment passées à
+d'autres fonctions. Pour définir une fonction anonyme en Elixir, nous avons besoin des mot-clés `fn` et `end`, à l'intérieur desquels nous pouvons définir
+n'importe quel nombre de paramètres et de corps de fonction, séparés par `->`.
+Jetons un coup d'œil à cet exemple :
```
Comment (tchoutri) [inline fr/lessons/basics/functions.md]: - `Jetons un coup d'œil à cet exemple :`
+ `Jetons un coup d'œil à cet exemple : `

File: fr/lessons/basics/functions.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+```elixir
+iex> sum = fn (a, b) -> a + b end
+iex> sum.(2, 3)
+5
+```
+
+### Le raccourci '&'
+
+Utiliser des fonctions anonymes est quelque chose de tellement courant en Elixir qu'il y a un raccourci pour le faire :
```
Comment (tchoutri) [inline fr/lessons/basics/functions.md]: - `raccourci pour le faire :`
+ `raccourci pour le faire : `

File: fr/lessons/basics/functions.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+iex> sum.(2, 3)
+5
+```
+
+Comme vous avez peut-être déjà deviné, dans la version raccourcie nos paramètres sont disponibles en tant que `&1`, `&2`, `&3`, etc.
+
+## Pattern matching
+
+Le Pattern matching (ou « Filtrage par motif » en Français) en Elixir ne se limite pas juste aux variables. Il peut être appliqué aux signatures de
+fonctions, comme nous allons le voir dans cette section :
```
Comment (tchoutri) [inline fr/lessons/basics/functions.md]: - `dans cette section :`
+ `dans cette section : `

File: fr/lessons/basics/functions.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+Elixir utilise le pattern matching pour identifier le groupe de paramètres correspondant, et appelle le corps de fonction correspondant :
```
Comment (tchoutri) [inline fr/lessons/basics/functions.md]: - `fonction correspondant :`
+ `fonction correspondant : `

File: fr/lessons/basics/functions.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+  def hello(name) do
+    "Hello, " <> name
+  end
+end
+
+iex> Greeter.hello("Sean")
+"Hello, Sean"
+```
+
+Si nos corps de fonction ne s'étendent que sur une ligne, nous pouvons les raccourcir encore plus avec le mot-clé `do:` :
```
Comment (tchoutri) [inline fr/lessons/basics/functions.md]: - `le mot-clé `do:` :`
+ `le mot-clé `do:` : `

**Comment by tchoutri** [comment]
… and so on.
Nobody's going to sue you if it's not like that right now but I'd need to keep this in mind. :)

**Comment by doomspork** [comment]
Ahh!  Extra whitespace, that I can remove.  I'm sorry, I thought you meant the punctuation itself was duplicated which I could not find.

**Comment by tchoutri** [comment]
No no no ! extra space that is *needed* after the `:` ! :D

**Comment by doomspork** [comment]
Haha, okay now I've got it 👍

**Pull request merged**